src/basic/unit-name.h

```diff
@@ -16,6 +15,18 @@ typedef enum UnitNameFlags {
         _UNIT_NAME_INVALID = -EINVAL,
 } UnitNameFlags;
 
+#define UNIT_NAME_MAX_LONG    PATH_MAX-1
+#define UNIT_NAME_MAX_DEFAULT 256
+
+static inline size_t unit_name_max_length(UnitType type) {
+    assert(type >= 0 && type < _UNIT_TYPE_MAX);
+    if (IN_SET(type, UNIT_DEVICE, UNIT_MOUNT, UNIT_SWAP))
+            /* We allow longer unit names for types where unit can get created because of external event
+             * outside of our control (e.g. device is connected or admin mounts a filesystem by hand). */
+            return UNIT_NAME_MAX_LONG;
+    return UNIT_NAME_MAX_DEFAULT;
```

Comment:
indetnation...

Reply:
why different limits for different types? that's awkward and unnecessary?

Reply:
The rationale for doing this just for mounts and devices: for those types the unit name can be specified externally and we just create a unit to match what the kernel tells us. For other types, the unit name is under our control. I think it makes sense to limit the change to only the specific unit types where we need this, but nowhere else. In particular, we prolly don't want service units with very long names.